### PR TITLE
Remove deprecated ReST API with dependencies

### PR DESCRIPTION
**Pull request by wbloszyk**
## Subject

Remove ReST API with dependencies and update routing with tests.
Remove DatagridBundle with replace Pager by custom Pager (to allow use news without AdminBundle)

I am targeting this branch, because this should be done in unstable branch.

### CODE REVIEW

**Review by VincentLanglet — COMMENTED** [review]
You should try to remove datagrid bundle too.

**Comment by wbloszyk** [comment]
> You should try to remove datagrid bundle too.

I think about it too. In NewsBundle we use pagination in frontend. We can remove datagrid bundle in favor of KnpPaginationBundle. I will wait with it for now.

**Comment by VincentLanglet** [comment]
> > You should try to remove datagrid bundle too.
> 
> I think about it too. In NewsBundle we use pagination in frontend. We can remove datagrid bundle in favor of KnpPaginationBundle. I will wait with it for now.

The DatagridBundle pagination was supposed to be only used for API.

For frontend, there is already a Pager/Datagrid class in the AdminBundle (and his persistence bundle), made for this.

**Comment by wbloszyk** [comment]
Ready to review. 
PHPStan and Psalm will added after merge #779 to 3.x and merge it to 4.x.

File: src/Action/ViewPostAction.php [hunk 1]
```diff
@@ -66,6 +66,8 @@ public function __invoke($permalink)
     {
         $post = $this->postManager->findOneByPermalink($permalink, $this->blog);
 
+        dump($permalink);
```
Comment (core23) [inline src/Action/ViewPostAction.php]: Debug code

**Review by core23 — CHANGES_REQUESTED** [review]

File: src/Pagination/BasePaginator.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    protected $pageSize;
+
+    /**
+     * @var \Traversable
+     */
+    protected $results;
+
+    /**
+     * @var
```
Comment (core23) [inline src/Pagination/BasePaginator.php]: Type is missing

**Review by core23 — CHANGES_REQUESTED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Review by core23 — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks

**Pull request merged**